### PR TITLE
Add migration to force removing not-null from existing tables because…

### PR DESCRIPTION
**Pull request by geofffranks**
… gorm has a bug and won't do it

- [x] Read the [Contributing document](../blob/-/.github/CONTRIBUTING.md).

Summary
---------------
Gorm has a bug and doesn't trigger any updates when running AutoMigrate to fix schema that is only removing `not null` from the column definition. We have to do this manually.

Backward Compatibility
---------------
Breaking Change? no

### CODE REVIEW

**Review by mariash — APPROVED** [review]

File: migration/v6/route.go [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+func (t *ModificationTag) Increment() {
+	t.Index++
+}
+
+func (m *ModificationTag) SucceededBy(other *ModificationTag) bool {
+	if m == nil || m.Guid == "" || other.Guid == "" {
+		return true
+	}
+
+	return m.Guid != other.Guid || m.Index < other.Index
```
Comment (andy-a-d-nguyen) [inline migration/v6/route.go:72]: **Question**

Can you explain what this function is supposed to do? I feel like a short comment would suffice.

File: migration/v6/route.go [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+		Index: 0,
+	}, nil
+}
+
+func (t *ModificationTag) Increment() {
+	t.Index++
+}
+
+func (m *ModificationTag) SucceededBy(other *ModificationTag) bool {
+	if m == nil || m.Guid == "" || other.Guid == "" {
```
Comment (andy-a-d-nguyen) [inline migration/v6/route.go:68]: **Question**

Can `other` ever be nil? Or is this a situation where we don't care if it is?

File: migration/v6/route.go (hunk 1 above)
Comment (andy-a-d-nguyen) [inline migration/v6/route.go:72]: **Question**

Would you ever run into a situation where both `m` and `other` are `nil` or `m` is not `nil`, but `other` is `nil`? If so, wouldn't this panic?

File: migration/v6/router_groups.go [hunk 3]
```diff
@@ -0,0 +1,284 @@
...
+			start: start,
+			end:   end,
+		}, nil
+	}
+	return Range{}, InvalidPortError
+}
+
+func (r Range) Overlaps(other Range) bool {
+	maxUpper := r.max(other)
+	minLower := r.min(other)
```
Comment (andy-a-d-nguyen) [inline migration/v6/router_groups.go:217]: **Question**

These are supposed to be integers that represent port numbers, correct? Would something more descriptive like `highestPort` and `lowestPort` or `upperLimit` or `lowerLimit` work?

File: migration/v6/tcp_route.go [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+	TcpMappingEntity
+}
+
+type TcpMappingEntity struct {
+	RouterGroupGuid string  `gorm:"not null; unique_index:idx_tcp_route" json:"router_group_guid"`
+	HostPort        uint16  `gorm:"not null; unique_index:idx_tcp_route; type:int" json:"backend_port"`
+	HostTLSPort     int     `gorm:"default:null; unique_index:idx_tcp_route; type:int" json:"backend_tls_port"`
+	HostIP          string  `gorm:"not null; unique_index:idx_tcp_route" json:"backend_ip"`
+	SniHostname     *string `gorm:"default:null; unique_index:idx_tcp_route" json:"backend_sni_hostname,omitempty"`
+	// We don't add uniqueness on InstanceId so that if a route is attempted to be created with the same detals but
```
Comment (andy-a-d-nguyen) [inline migration/v6/tcp_route.go:22]: **Question**

Did you mean `details` here?

**Pull request merged**